### PR TITLE
Adding tests for ffead-cpp (mongodb, mysql, postgresql) on

### PR DESCRIPTION
**Pull request by sumeetchhetri**
- Embedded server
- Apache
- Nginx
Resolves https://github.com/TechEmpower/FrameworkBenchmarks/issues/2554

### CODE REVIEW

**Comment by nbrady-techempower** [comment]
@sumeetchhetri ffead was removed because the test implementations did not live in the repo. This does not do anything to address that. See issue #2554

**Comment by sumeetchhetri** [comment]
Done as suggested, Please refer implementation files at https://github.com/sumeetchhetri/FrameworkBenchmarks/tree/master/frameworks/C%2B%2B/ffead-cpp/te-benchmark
Refer usage in the setup script at https://github.com/sumeetchhetri/FrameworkBenchmarks/blob/master/toolset/setup/linux/frameworks/ffead-cpp-framework.sh (line 12/13)

**Comment by sumeetchhetri** [comment]
Finally all tests are successful

**Comment by nbrady-techempower** [comment]
Thanks @sumeetchhetri ! I will get around to reviewing this today or tomorrow. If it's all good, it will make it for Round 15 Preview 2

File: toolset/setup/linux/frameworks/ffead-cpp-apache-mongo.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+rm -rf web/default web/oauthApp web/flexApp web/markers
+chmod 755 *.sh resources/*.sh rtdcf/autotools/*.sh
+./server.sh > ffead-cpp-apache-mongo.log 2>&1
+while ! echo exit | nc localhost 8080; do sleep 5; done
+rm -f serv.ctrl
+sleep 10
+
+cd ${IROOT}
+
+touch ${IROOT}/ffead-cpp-apache-mongo.installed
```
Comment (msmith-techempower) [inline toolset/setup/linux/frameworks/ffead-cpp-apache-mongo.sh]: This line should be updated to set up the environment when used. For example, the line in the setup scripts:
`${IROOT}/httpd/bin/apachectl restart > ffead.log 2>&1`
should be
`apachectl restart > ffead.log 2>&1`

So, this line should be something like
`echo -e "export PATH=${IROOT}/httpd/bin:\$PATH" > $IROOT/ffead-cpp-apache-mongo.installed`

File: toolset/setup/linux/frameworks/ffead-cpp-apache-mongo.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+cp -f web/te-benchmark/config/sdormmongo.xml web/te-benchmark/config/sdorm.xml
+rm -rf ffead-cpp-2.0-bin
+make build-apps
+rm -rf ${IROOT}/ffead-cpp-2.0
+cp -rf ffead-cpp-2.0-bin ${IROOT}/ffead-cpp-2.0
+cd ${IROOT}/ffead-cpp-2.0
+rm -rf web/default web/oauthApp web/flexApp web/markers
+chmod 755 *.sh resources/*.sh rtdcf/autotools/*.sh
+./server.sh > ffead-cpp-apache-mongo.log 2>&1
+while ! echo exit | nc localhost 8080; do sleep 5; done
```
Comment (msmith-techempower) [inline toolset/setup/linux/frameworks/ffead-cpp-apache-mongo.sh]: This loop is unnecessary; please remove.

Comment (sumeetchhetri) [inline toolset/setup/linux/frameworks/ffead-cpp-apache-mongo.sh]: ffead-cpp generates a lot of dynamic code on initial startup, which is where it generates and compiles stuff on the fly and creates a dynamic library. This is a one time init step which is very much required.

**Pull request merged**